chat-plugins/panagrams-beta.js

```diff
@@ -152,4 +152,135 @@ exports.commands = {
 					"The type of this Pokemon is: <b>" + Tools.getTemplate(room.panagram.answer).types + "</b>";
 		this.sendReplyBox("Panagram hint: <br />The Pokemon is <b>" + answer_length + "</b> characters long.  <br />" + hint);
 	}
-};
\ No newline at end of file
+};*/
+
+//Panagrams by SilverTactic (Siiilver) and panpawn
+//#Tyme4BugHunting
+var pGames = exports.pGames = {};
+
+function mix(word) {
+	var arr = [];
+	for (var k = 0; k < word.length; k++) {
+		arr.push(word[k]);
+	}
+	var a, b, i = arr.length;
+	while (i) {
+		a = Math.floor(Math.random() * i);
+		i--;
+		b = arr[i];
+		arr[i] = arr[a];
+		arr[a] = b;
+	}
+	return arr.join('').toString();
+}
+
+var Panagram = (function () {
+	function Panagram(room, category) {
+		this.room = room;
+		this.category = category;
+		switch (toId(category)) {
+		case 'items':
```

Comment:
Unlike main, I like to indent my switch cases.  So, from this line out, indent one in the entire switch case.